beeflow/task_manager/background.py

```diff
@@ -70,6 +72,11 @@ def update_jobs(db):
         task = job.task
         job_id = job.job_id
         job_state = job.job_state
+
+        if job_state in COMPLETED_STATES:
+            db.job_queue.remove_by_id(id_)
+            continue
```

Comment:
I don't understand the use of continue here, if you remove the task from the queue should it check for state updates?

Reply:
My thought process was that if a task is completed, it should be removed from the queue and then the "for job in job_q" loop should move onto the next job. The continue skips the check state updates and moves onto the next job. I was hoping this could fix the race condition because if a job is in a completed state it doesn't check for updates anymore.